fix: JSON form (#1049)
appsmithorg/appsmith-docs: website/docs/reference/widgets/json-form.md

You can display dynamic data in a JSON Form widget by binding the response from a query or a JS function to the **Source Data** property. This allows the form to update dynamically as the data changes in the database or API.

---

For example, you want to create JSON form using [sample users database](/core-concepts/connecting-to-data-sources/connecting-to-databases#sample-databases), and retrieving the necessary data from a query named `fetchUserData`. You can display the fetched data by binding the query response to the Table Data property of the Table widget `tblUserData`, as shown below:

```js
{{fetchUserData.data}}
```

To automatically generate the fields in the JSON Form when a table row is selected, add the below code in the **Source Data** property:

```js
{{tbluserData.selectedRow}}
```

You can click on an individual row in the Table and update data in the form fields.

#### Auto generate form

> Enabling the **Auto Generate Form** property can save time and effort when dealing with changing source data. This feature ensures that the form fields are updated automatically based on any changes in the data, such as changes in keys or data types.

Reviewer: ```suggestion
You can enable the **Auto Generate Form** property to have the form fields regenerate automatically when the source data changes, for example, when keys in the JSON change or if a data type changes, for example from string to a number. 
```

> However, it's important to note that enabling this feature can override any custom configurations that you may have made using JavaScript for data transformation. 

Reviewer: ```suggestion
However, it's important to note that enabling this feature overrides any custom configurations that you are providing through data transformations using JavaScript.
```

#### Field configuration

> In the field configuration property, you can add or update different fields for the JSON form. The available fields that can be configured include:

Reviewer: ```suggestion
In the **Field Configuration** section of the JSON Form property pane, you can add fields or update each field's settings by clicking the cog icon ⚙︎ next to it.  You can customize each form field using properties, for instance updating the field type, setting validation and style properties, and triggering actions using event listeners. You can select any one of the following options from the **Field Type** property to update the widget type for that field on the JSON Form:
```

* [Source data exceeds 50 fields](/help-and-support/troubleshooting-guide/widget-errors#source-data-exceeds-50-fields)

> If you run into any other issues while working with the widget, check out the guide on [widget errors guide](/help-and-support/troubleshooting-guide/widget-errors). If the guide doesn't cover your issue,  please connect with **support@appsmith.com** or raise your query on the [Discord Server](https://discord.com/invite/rBTTVJp).

Reviewer: ```suggestion
If you encounter other issues while working with the widget, see [widget errors](/help-and-support/troubleshooting-guide/widget-errors). If the guide doesn't cover your issue, contact the [support team](mailto:support@appsmith.com).
```
